pkg/bootstrap/init.go

```diff
@@ -22,105 +22,129 @@ import (
 		return err
 	}
 
-	if err := createOrUpdateFromYaml(subscription, client, reader); err != nil {
+	klog.Info("create extra resources for common services")
+	if err := createOrUpdateResources(annotations, strings.Split(annotations[CsExtResource], ","), client, reader); err != nil {
 		return err
 	}
 
-	// create extra yamls
-	klog.Info("create extra yaml resources")
-	if err := createOrUpdateResourcesFromDir(filepath.Join(resourcesDir, "extra"), client, reader); err != nil {
+	klog.Info("create ODLM  OperandRegistry and OperandConfig CR resources")
+	if err := waitResourceReady(config, "operator.ibm.com/v1alpha1", "OperandRegistry"); err != nil {
 		return err
 	}
-
-	// create operandConfig and operandRegistry
-	klog.Info("create OperandConfig and OperandRegistry")
-	operandConfig, err := ioutil.ReadFile(filepath.Join(resourcesDir, "cs-operandconfig.yaml"))
-	if err != nil {
+	if err := waitResourceReady(config, "operator.ibm.com/v1alpha1", "OperandConfig"); err != nil {
 		return err
 	}
-	operandRegistry, err := ioutil.ReadFile(filepath.Join(resourcesDir, "cs-operandregistry.yaml"))
-	if err != nil {
+	if err := createOrUpdateResources(annotations, OdlmCrResources, client, reader); err != nil {
 		return err
 	}
 
-	timeout := time.After(300 * time.Second)
-	ticker := time.NewTicker(30 * time.Second)
-	for {
-		klog.Info("try to create IBM Common Services OperandConfig and OperandRegistry")
-		select {
-		case <-timeout:
-			return fmt.Errorf("timeout to create the ODLM resource")
-		case <-ticker.C:
-			// create OperandConfig
-			errConfig := createOrUpdateFromYaml(operandConfig, client, reader)
-			if errConfig != nil {
-				klog.Error("create OperandConfig error with: ", errConfig)
-			}
+	return nil
+}
 
-			// create OperandRegistry
-			errRegistry := createOrUpdateFromYaml(operandRegistry, client, reader)
-			if errRegistry != nil {
-				klog.Error("create OperandRegistry error with: ", errRegistry)
+func createOrUpdateResources(annotations map[string]string, resNames []string, client client.Client, reader client.Reader) error {
+	for _, res := range resNames {
+		if r, ok := annotations[res]; ok {
+			klog.Infof("create resource: %s", res)
+			if err := createOrUpdateFromYaml([]byte(r), client, reader); err != nil {
+				return err
 			}
+		} else {
+			klog.Warningf("no resource %s found in annotations", res)
+		}
+	}
+	return nil
+}
 
-			if errConfig == nil && errRegistry == nil {
-				return nil
-			}
+func getDeployment(reader client.Reader) (*appsv1.Deployment, error) {
+	deploy := &appsv1.Deployment{}
+	deployName, err := k8sutil.GetOperatorName()
+	if err != nil {
+		return nil, fmt.Errorf("could not find the operator name: %v", err)
+	}
+	deployNs, err := k8sutil.GetOperatorNamespace()
+	if err != nil {
+		return nil, fmt.Errorf("could not find the operator namespace: %v", err)
+	}
+	err = reader.Get(context.TODO(), types.NamespacedName{Name: deployName, Namespace: deployNs}, deploy)
```

Comment:
@DanielXLee Let's add a loop to periodically get the deploy 3 times, interval 1 minute.

Reply:
added @chenzhiwei